src/App.tsx

```diff
@@ -15,36 +16,87 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  Default = 'default',
+  Alphabetical = 'alphabetical',
+  Length = 'length',
+}
+
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>([...goodsFromServer]);
+  const [sortField, SetSortField] = useState<SortType>(SortType.Default);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const handleSort = (sortType: SortType) => {
+    const sortedGoods = [...goodsFromServer];
+
+    if (sortType === SortType.Alphabetical) {
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+    } else if (sortType === SortType.Length) {
+      sortedGoods.sort((a, b) => a.length - b.length);
+    }
+
+    setGoods(isReversed ? sortedGoods.reverse() : sortedGoods);
+    SetSortField(sortType);
+  };
```

Comment:
- You reverse in the sort handler, it's confusing
- Use switch instead of `if else`

To simplify just create a function outside the component that will prepare goods. Store them in a variable and just render them, no need to store them in a state as it's a dynamic value

```typescript
const preparedGoods = prepareGoods(goods, { isReversed, sortType });
```